prometheus/cache.go

```diff
@@ -36,6 +37,9 @@ type (
 	}
 
 	promCacheImpl struct {
+		// to hold internal data such as the logger
+		ctx context.Context
```

Comment:
Again I think it'd be better to save the logger directly rather than the ctx on the struct.

Reply:
I changed it to store the logger, and in fact I think we'll eventually want to remove that as well. I think we will likely want to change the cache functions to take a context and we just pass the logger along inside of that. For now I just added a TODO in the code for us to look into that in the future.